examples/jetstream/api-migration/go/main.go

```diff
@@ -0,0 +1,255 @@
+	// In order to create an entry point to the JetStream API, use the
+	// `New` function.
+	newJS, _ := jetstream.New(nc)
+
+	// The new API uses `context.Context` for cancellation and timeouts when
+	// managing streams and consumers.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Creating a stream is done using the `CreateStream` method.
+	// It works similarly to the legacy `AddStream` method, except
+	// instead of returning `StreamInfo`, it returns a `Stream` handle,
+	// which can be used to manage the stream.
+	// Instead of creating a new stream, let's look up the existing `EVENTS` stream.
+	stream, _ := newJS.Stream(ctx, streamName)
+
+	// The new API differs from the legacy API in that it does not
+	// auto-create consumers. Instead, consumers must be created or retrieved
+	// explicitly. This allows for more control over the consumer lifecycle,
+	// while also getting rid of the hidden logic of the `Subscribe()` methods.
+	// In order to create a consumer, use the `AddConsumer` method.
+	// This method works similarly to the legacy `AddConsumer` method,
+	// except it returns a `Consumer` handle, which can be used to manage
+	// the consumer. Notice that since we are using pull consumers, we
+	// do not need to provide a `DeliverSubject`.
+	// In order to create a short-lived, ephemeral consumer, we will set the
+	// `InactivityThreshold` to a low value and not provide a consumer name.
+	cons, _ := stream.AddConsumer(ctx, jetstream.ConsumerConfig{
+		InactiveThreshold: 10 * time.Millisecond,
+	})
+	fmt.Println("Created consumer", cons.CachedInfo().Name)
+
+	// ### Continuous message retrieval with `Consume()`
+	// In order to continuously receive messages, the `Consume` method
+	// can be used. This method works similarly to the legacy `Subscribe`
+	// method, in that it will asynchronously deliver messages to the
+	// provided `jetstream.MsgHandler` function. However, it does not
+	// create a consumer, instead it will use the consumer created
+	// previously.
+	fmt.Println("# Consume messages using Consume()")
+	consumeContext, _ := cons.Consume(func(msg jetstream.Msg) {
+		fmt.Printf("received %q\n", msg.Subject())
+		msg.Ack()
+	})
+	time.Sleep(100 * time.Millisecond)
+
+	// `Consume()` returns a `jetstream.ConsumerContext` which can be used
+	// to stop consuming messages. In contrast to `Unsubscribe()` in the
+	// legacy API, this will not delete the consumer.
+	// Consumer will be automatically deleted by the server when the
+	// `InactivityThreshold` is reached.
+	consumeContext.Stop()
+
+	// Now let's create a new, long-lived, named consumer.
+	// In order to filter messages, we will provide a `FilterSubject`.
+	// This is equivalent to providing a subject to `Subscribe` in the
+	// legacy API.
+	consumerName = "pull-1"
+	cons, _ = stream.AddConsumer(ctx, jetstream.ConsumerConfig{
+		Name:              consumerName,
+		InactiveThreshold: 10 * time.Minute,
```

Comment:
Maybe note that an inactive threshold is not necessary for a durable, but can be set if there is some forced upper bound desired.